### PR TITLE
[Test Proxy] Maps Render migration

### PR DESCRIPTION
**Pull request by vcolin7**
Updated test classes to use Test Proxy and migrated recordings to the assets repo. Also had to modify the base test class to improve `HttpPipeline` creation for test clients.

### CODE REVIEW

File: sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java [hunk 1]
```diff
@@ -52,60 +61,67 @@ protected void beforeTest() {
...
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```
Comment (samvaity) [inline sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java:81]: This should already be redacting, is it not?

File: sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java [hunk 2]
```diff
@@ -52,60 +61,67 @@ protected void beforeTest() {
...
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode()) {
```
Comment (samvaity) [inline sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java:78]: Sanitizers are to be added in both Record and Playback mode.
image

**Review by samvaity — APPROVED** [review]
Other changes lgtm 🎉

File: sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java (hunk 1 above)
Comment (vcolin7) [reply sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java:81]: I don't think we have released the new version of Core Test that does this. Should I make the libraries use it even if it's using the current version from source?

Comment (vcolin7) [reply sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java:81]: I'll change the code once the new version of Core Test is released.

**Pull request merged**